src/rivian/rivian_ble.py

```diff
@@ -0,0 +1,85 @@
+import platform
+import asyncio
+import secrets
+from asyncio import TimeoutError
+from bleak import BleakScanner, BleakClient, BleakError
+from rivian import utils
+
+RIVIAN_PHONE_ID_VEHICLE_ID_UUID = "AA49565A-4D4F-424B-4559-5F5752495445"
+RIVIAN_PNONCE_VNONCE_UUID = "E020A15D-E730-4B2C-908B-51DAF9D41E19"
+RIVIAN_BLE_ACTIVE_ENTRY_UUID = "5249565F-4D4F-424B-4559-5F5752495445"
+RIVIAN_PHONE_KEY_LOCAL_NAME = "Rivian Phone Key"
+
+# Create an asyncio.Event object to signal the arrival of a new notification.
+notification_event = asyncio.Event()
+notification_data = None
+
+def notification_handler(sender, data):
+    global notification_data
+    notification_data = data
+    notification_event.set()
+    
+async def scan_for_device(target_device_name): 
+    while True:
+        # Sleep before the next scan
+        await asyncio.sleep(1)  
+
+        devices = await BleakScanner.discover()
+        for device in devices:
+            if device.name is not None and target_device_name in device.name:
+                print(f"Found Device: {device.name}, Address: {device.address}")
+                return device.address
+
+async def connect_to_device(address):
+    global client
+    print(f"Connecting to {address}")
+
+    try:
+        client = BleakClient(address, timeout=10.0)
+        await client.connect()
+        print(f"Connected: {client.is_connected}")
+
+        return client.is_connected 
+    except (BleakError, TimeoutError, OSError):
+        print(f"Failed to connect to {address}. Retrying...")
+        return False 
```

Comment:
We shouldn't use globals here either and bleak recommends against trying to open a BleakClient by address as it causes issues (see the warning at https://bleak.readthedocs.io/en/latest/api/client.html#bleakclient-class). Instead we should use a context manager in the `pair_phone` method below:
```python
async with BleakClient(device, timeout=10) as client:
    ...
```